lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -139,60 +135,47 @@ def f(directory: pathlib.Path, tar_file: pathlib.Path):
                 args=[], returncode=returncode, stdout=stdout, stderr=None
             )
 
-        TestSendDirectory.function_called.append("fake_create_tar")
+        __class__.functions_called.append("fake_create_tar")
         return f
 
-    @staticmethod
-    def fake_unlink(*args):
-        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
-        TestSendDirectory.function_called.append("fake_unlink")
-        pass
+    @responses.activate
+    def test_tar_create_success(self, tool_meister, monkeypatch):
+        """This test should pass the tar creation in send directory"""
 
-    @staticmethod
-    def fake_open(*args):
-        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
-        TestSendDirectory.function_called.append("fake_open")
-        return io.StringIO()
+        def fake_unlink(*args):
+            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_unlink")
 
-    def fake_rmtree(self, directory: pathlib.Path):
-        assert directory == tmp_dir
-        self.function_called.append("fake_rmtree")
-        pass
+        def fake_open(*args):
+            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_open")
+            return io.StringIO()
```

Comment:
Why do we need to do this?  In Python a function invocation is the same.  The mock'd function just needs to verify what it is expected to be called with, so this is just syntactic sugar, no?

Reply:
I was prepared to embrace your argument until earlier when Nikhil suggested having evidence that Python's calling conventions are not as simple as your prompt here suggests.

Also, I think it makes it easier for the reader if the thing being called closely parallels what the caller expects it to be.  (And, it's not hard to wrap these two functions in an inner class.)

The better question is, should we be refactoring the beautiful `MockedPath` class that you (@portante) have implemented to make it available for use here?

Reply:
Hmm, I see that @portante has implemented the `MockedPath` class in his PR #2725, I could implement something similar here (which will eventually gets merged/rebased on Peter's changes anyway) or I could keep it as it is for now and let peter add these in his `MockedPath` class when doing the rebase :upside_down_face:

Reply:
I would vote for implementing something similar here now, rather than counting on the future work to remember to address this.  I think what is needed for this PR is pretty simple.

Reply:
I have to do a bit of surgery to merge with PR #2725, so I can add that there and track the work during the integration.

Reply:
Okie....